dopamine/utils/lock.py

```diff
@@ -0,0 +1,90 @@
+# coding=utf-8
+# Copyright 2018 The Dopamine Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Creates a function decorator to protect execution with a lock."""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import functools
+import threading
+
+_DEFAULT_LOCK_ATTR_NAME = '_lock'
+
+
+# TODO(aarg): Add a lock implementation that lets read operation occur at the
+# same time (i.e. less restrictive).
+
+
+def initialize_lock(object_to_lock, lock=threading.Lock(),
```

Comment:
This is a nice helper method.

Reply:
re: https://github.com/wes-turner/dopamine/pull/3#discussion_r253893966 , I like your decision.